firefox-ios/Client/Frontend/InternalSchemeHandler/ErrorPageHelper.swift

```diff
@@ -153,11 +153,33 @@ class ErrorPageHandler: InternalSchemeResponse, FeatureFlaggable {
     // When nativeErrorPage feature flag is true, only create
     // html page with gray background similar to homepage or privatehomepage.
     // TODO: responseForErrorWebPage() will be removed in future with rest of the old error page code.
+    var isNativeErrorPageEnabled: Bool {
+        return featureFlags.isFeatureEnabled(.nativeErrorPage, checking: .buildOnly)
+    }
+
+    var isNICErrorPageEnabled: Bool {
+        return featureFlags.isFeatureEnabled(.noInternetConnectionErrorPage, checking: .buildOnly)
+    }
+
     func response(forRequest request: URLRequest) -> (URLResponse, Data)? {
-        if featureFlags.isFeatureEnabled(.nativeErrorPage, checking: .buildOnly) {
-            responseForNativeErrorPage(request: request)
+        guard let url = request.url,
+              let components = URLComponents(url: url, resolvingAgainstBaseURL: false),
+              let code = components.valueForQuery("code"),
+              let errCode = Int(code) else {
+            return nil
+        }
+
+        /// Used for checking if current error code is for no internet connection
+        let noInternetErrorCode = Int(
+            CFNetworkErrors.cfurlErrorNotConnectedToInternet.rawValue
+        )
+
+        if isNativeErrorPageEnabled {
+            return responseForNativeErrorPage(request: request)
+        } else if isNICErrorPageEnabled && (errCode == noInternetErrorCode) {
+            return responseForNativeErrorPage(request: request)
         } else {
```

Comment:
Hmmm, I'm concerned about this check. If we enabled native error pages, we will NEVER show your no internet connection error page, right? 🤔 

I think we need a little bit more logic here to determine the type of error pages we need to show, and THEN we can decide if we are allowed to show them (isNativeErrorPageEnabled, isNICErrorPageEnabled)

Reply:
Oh, hold on, I just realized both these paths call the same function!

So maybe the if statements should just be combined in that case to be clear they are the same code paths? 🤔

Reply:
Good question!
If isNativeErrorPageEnabled = true then all the new error pages can be shown based on the error code (including "no internet" page). But if isNICErrorPageEnabled = true and error code is noInternetErrorCode, then only new "no internet error" page is shown. The current plan is to get No Internet Error Pages in v134, hence this temporary flag.

Reply:
Ahh I understand, thank you @tusharC95! This totally makes sense. 😄  You can disregard my comment here. 👍 

So the no internet error pages are going to be released first in v134, and the new error pages will come in a later release?

Reply:
Yes, that's the plan. 🤞

Reply:
Awesome, thank you for filling me in!